pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3694,7 +3694,7 @@ private Topic getTopicReference(TopicName topicName) {
             throw e;
         } catch (Exception e) {
             if (e.getCause() instanceof NotAllowedException) {
-                throw new RestException(Status.CONFLICT, e.getCause());
+                throw new RestException(Status.BAD_REQUEST, e.getCause());
```

Comment:
Interesting, do we through `NotAllowedException` for `pulsar().getBrokerService().getTopicIfExists()`? @congbobo184 Could you please help check how to improve this part? From the method name, we only get the topic ref if the topic exists, but looks like it also create topics.

Reply:
ok

Reply:
+1